parsl/executors/workqueue/executor.py

```diff
@@ -321,6 +321,32 @@ def submit(self, func, resource_specification, *args, **kwargs):
         kwargs : dict
             Keyword arguments to the Parsl app
         """
+        cores = None
+        memory = None
+        disk = None
+        if resource_specification and isinstance(resource_specification, dict):
+            logger.debug("Got resource specification: {}".format(resource_specification))
+
+            acceptable_resource_types = ['cores', 'memory', 'disk']
+            keys = list(resource_specification.keys())
+            if len(keys) != 3:
+                logger.error("Task resource specification requires "
+                             "three resources to be specified simultaneously: cores, memory, and disk")
```

Comment:
Why do we require all three to be specified? @btovar should correct me if I'm wrong, but I think we can allow users to provide a guess for one but not the others?

Reply:
Testing this PR, I see that if it's not specified, WQ will ask for everything available, which will usually result in one task/worker. Users expecting the constraint to only apply to the specified resource will be surprised. So in light of that, I think it makes sense to require all three to be specified, as implemented here.